x-pack/plugins/serverless_observability/public/components/side_navigation/index.tsx

```diff
@@ -13,235 +13,250 @@ import {
-                return regex.test(pathNameSerialized);
+            ],
+          },
+          {
+            id: 'apm',
+            title: i18n.translate('xpack.serverlessObservability.nav.applications', {
+              defaultMessage: 'Applications',
+            }),
+            renderAs: 'accordion',
+            children: [
+              {
+                link: 'apm:services',
+                getIsActive: ({ pathNameSerialized }) => {
+                  const regex = /app\/apm\/.*service.*/;
+                  return regex.test(pathNameSerialized);
+                },
               },
-            },
-            {
-              link: 'apm:traces',
-              getIsActive: ({ pathNameSerialized, prepend }) => {
-                return pathNameSerialized.startsWith(prepend('/app/apm/traces'));
+              {
+                link: 'apm:traces',
+                getIsActive: ({ pathNameSerialized, prepend }) => {
+                  return pathNameSerialized.startsWith(prepend('/app/apm/traces'));
+                },
               },
-            },
-            {
-              link: 'apm:dependencies',
-              getIsActive: ({ pathNameSerialized, prepend }) => {
-                return pathNameSerialized.startsWith(prepend('/app/apm/dependencies'));
+              {
+                link: 'apm:dependencies',
+                getIsActive: ({ pathNameSerialized, prepend }) => {
+                  return pathNameSerialized.startsWith(prepend('/app/apm/dependencies'));
+                },
               },
-            },
-          ],
-        },
-        {
-          id: 'metrics',
-          title: i18n.translate('xpack.serverlessObservability.nav.infrastructure', {
-            defaultMessage: 'Infrastructure',
-          }),
-          renderAs: 'accordion',
-          children: [
-            {
-              link: 'metrics:inventory',
-              getIsActive: ({ pathNameSerialized, prepend }) => {
-                return pathNameSerialized.startsWith(prepend('/app/metrics/inventory'));
+            ],
+          },
+          {
+            id: 'metrics',
+            title: i18n.translate('xpack.serverlessObservability.nav.infrastructure', {
+              defaultMessage: 'Infrastructure',
+            }),
+            renderAs: 'accordion',
+            children: [
+              {
+                link: 'metrics:inventory',
+                getIsActive: ({ pathNameSerialized, prepend }) => {
+                  return pathNameSerialized.startsWith(prepend('/app/metrics/inventory'));
+                },
               },
-            },
-            {
-              link: 'metrics:hosts',
-              getIsActive: ({ pathNameSerialized, prepend }) => {
-                return pathNameSerialized.startsWith(prepend('/app/metrics/hosts'));
+              {
+                link: 'metrics:hosts',
+                sideNavStatus: settings.hasInfrastructureHosts ? 'visible' : 'hidden',
```

Comment:
Not sure if you have tried it, but I think it might be simpler if we'd keep this static, but hide/show the link by updating the `metrics:hosts` `navLinkStatus` in core:

https://github.com/elastic/kibana/blob/9a8277e26086f8b89389e4eeba5c8e375d94f112/x-pack/plugins/infra/public/plugin.ts#L270 , it should be possible to update this visibility 

The serverless side nav should hide the link from side nav based on that visibility.

Reply:
Oh, you're right, thank you!. And it actually already works because of [this](https://github.com/elastic/kibana/blob/9a8277e26086f8b89389e4eeba5c8e375d94f112/x-pack/plugins/infra/public/plugin.ts#L339), not sure how I missed this 🙈